### PR TITLE
Joined Lists Pipeline Source

### PR DESCRIPTION
**Pull request by haspadar**
- Added JoinedLists DSL source op that emits the left-major cartesian product of two list settings joined by a separator
- Updated DEV.md with the new op entry and a usage example

Closes #768

## Release Notes

* **New Features**
  * Added `JoinedLists` operation to combine two configuration lists using a custom separator, generating a left-major product with empty results when either input is empty.

* **Documentation**
  * Added DSL documentation and usage examples for the new list-joining operation.

* **Tests**
  * Added comprehensive test coverage validating joins, cartesian products, edge cases, and error handling.

[Review Change Stack](https://app.coderabbit.ai/change-stack/haspadar/sheriff/pull/769?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by haspadar** [comment]
@coderabbitai review

File: src/Chain/Parse/JoinedListsFormula.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     */
+    private function pairs(array $left, array $right, string $separator): array
+    {
+        $result = [];
+
+        foreach ($left as $leftPart) {
+            foreach ($right as $rightPart) {
+                $result[] = new StringValue(sprintf('%s%s%s', $leftPart->raw, $separator, $rightPart->raw));
+            }
+        }
```
[the remark this answers is not in the document]
Comment (haspadar) [reply src/Chain/Parse/JoinedListsFormula.php:127]: Declining: cartesian semantics is the intended behavior (issue updated to match). Empty on either side yields no items by design — symmetry beats issue-text wording. @coderabbitai resolve
